src/Platforms/SQLServer2012Platform.php

```diff
@@ -504,7 +513,17 @@ private function _appendUniqueConstraintDefinition($sql, Index $index)
             $fields[] = $field . ' IS NOT NULL';
         }
 
-        return $sql . ' WHERE ' . implode(' AND ', $fields);
+        if (count($fields) > 0) {
```

Comment:
this, is much, much harder to read than before

Reply:
I will check this. At least the check for partial index support can be removed now.

Reply:
Fixed